graveler/prefix.go

```diff
@@ -0,0 +1,17 @@
+package graveler
+
+import "math"
+
+func UpperBoundForPrefix(prefix []byte) []byte {
+	idx := len(prefix) - 1
+	for idx >= 0 && prefix[idx] == math.MaxUint8 {
+		idx--
+	}
+	if idx == -1 {
+		return nil
```

Comment:
Will add code doc for this one.
About the append(prefix, 0) in this case, this function returns nil because it wanted the caller to handle the case where there is no value which we can use the prefix size to upper bound the range.